### PR TITLE
CI: missing jobs on changes in components package.

### LINKED ISSUE #47347: ci-jobs is not returning tests for some dependencies changes
ci-jobs utility should return tests jobs for core when only dependencies of any level are changed.
This doesn't happen with `@woocommerce/components`, which is a dependency of `admin-library`, which is a dependency of woocommerce plugin. The expectation is that a change in `@woocommerce/components` would trigger woocommerce plugin's tests.

Note: it works as expected with other dependencies, like `@woocommerce/product-editor`.

### Additional context

https://github.com/woocommerce/woocommerce/pull/43345
https://github.com/woocommerce/woocommerce/pull/47271

### Dev-notes

We are not going into full-scale refactoring here - aiming to fix for the specified case only.

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/47347.

The change expands https://github.com/woocommerce/woocommerce/pull/47271 a bit and adds detecting dependency change based on available changelist.

### Testing

Validate the following jobs run https://github.com/woocommerce/woocommerce/actions/runs/9872263649/job/27262049639?pr=49330 - note dependant packages linting/testing is being executed (incl. WooCommerce itself).

The change was made and later reverted in the components JS package.

### CODE REVIEW

File: packages/js/components/src/utils.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ type ChildrenProps = {
 
 /**
  * Returns an object with the children and props that will be used by `cloneElement`. They will change depending on the
- * type of children passed in.
+ * type of children passed in..
```
Comment (kalessil) [inline packages/js/components/src/utils.tsx]: To be reverted before merge.

File: packages/js/components/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "@woocommerce/components",
 	"version": "12.3.0",
-	"description": "UI components for WooCommerce.",
+	"description": "UI components for WooCommerce...",
```
Comment (kalessil) [inline packages/js/components/package.json]: To be reverted before the merge.

**Comment by kalessil** [comment]
@adimoldovan: should not this changes deprecate `Blocks e2e - trigger for Blocks e2e tests in @woocommerce/plugin-woocommerce` job from https://github.com/woocommerce/woocommerce/pull/48224?

**Comment by adimoldovan** [comment]
> @adimoldovan: should not this changes deprecate `Blocks e2e - trigger for Blocks e2e tests in @woocommerce/plugin-woocommerce` job from #48224?

I think so, yeah. Let's test it by removing it and changing something in the blocks e2e tests.

File: tools/monorepo-utils/src/ci-jobs/lib/job-processing.ts [hunk 3]
```diff
@@ -302,13 +302,13 @@ async function createJobsForProject(
...
-				( array ) => array.length > 0
-			)
-		) {
-			dependencyChanges = true;
+		if ( dependencyChanges === false ) {
+			// First line of detection: implicit changes list points to the dependency.
+			dependencyChanges = ( changes[ dependency.name ] || [] ).length > 0;
+			if ( dependencyChanges === false ) {
+				// Second line of detection: the dependency spawns testing jobs.
+				dependencyChanges = dependencyJobs.test.length > 0;
```
Comment (adimoldovan) [inline tools/monorepo-utils/src/ci-jobs/lib/job-processing.ts]: On the second line of detection, I think we should include any job type, including lint.

Comment (kalessil) [reply tools/monorepo-utils/src/ci-jobs/lib/job-processing.ts]: Addressed in 7224d30a4a94723046547d5c9fa6c68dfe054112

**Comment by kalessil** [comment]
@adimoldovan: addressed your feedback and ready for the final review round.

**Review by adimoldovan — APPROVED** [review]
Looks good.

**Pull request merged**